src/generic/complex_matrices.cc

```diff
@@ -899,8 +899,8 @@ namespace oomph
       }
 
       // build
-      unsigned long this_nnz = this->nnz();
-      result.build_without_copy(value, column_index, row_start, this_nnz, m, n);
+      unsigned long nnz = this->nnz();
```

Comment:
Change to `n_row` as previously requested(!).

You should **never** define a variable that has the same name as a function otherwise you will overwrite it in the current scope

**Example:**

```cpp
#include <iostream>

struct A
{
    void f()
    {
        // Redefine g in local scope 
        int g = 0;

        // Works
        std::cout << g << std::endl;

        // Error!
        std::cout << g() << std::endl;
    }
    
    int g() { return -1; }
};

int main()
{
    A a;
    a.f();
}

// Output
In member function 'void A::f()':
9:24: error: 'g' cannot be used as a function
```